### 1.Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
### zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
### ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
### Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy. Полученные значения должны быть равны. Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

In [1]:
import numpy as np
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
cov=np.mean(zp*ks)-np.mean(zp)*np.mean(ks)
cov

9157.839999999997

In [2]:
np.cov(zp,ks,ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [3]:
np.std(zp)

59.115480206118605

In [4]:
np.std(ks)

174.55340157098058

In [5]:
9157.839999999997/(59.115480206118605*174.55340157098058)

0.8874900920739158

In [6]:
np.corrcoef(zp,ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

_______________________________________________________________________________________________________________________________

### 2.Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:
### 131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
### Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [7]:
from statistics import mean
x=[131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
avg = mean(x)
q=np.std(x, ddof=1)
i1=avg+2.262*q/10**0.5
i2=avg-2.262*q/10**0.5
print(f'Искомый довирительный интервал: [{i2};{i1}]')

Искомый довирительный интервал: [110.55660776308164;125.64339223691834]


Ответ: Искомый довирительный интервал: [110.55660776308164;125.64339223691834]

_______________________________________________________________________________________________________________________________

### 3. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [8]:
i1=174.2+1.96*25/27**0.5
i2=174.2-1.96*25/27**0.5
print(f'Искомый довирительный интервал: [{i2};{i1}]')

Искомый довирительный интервал: [164.7699456032361;183.63005439676388]


Ответ: Искомый довирительный интервал: [164.7699456032361;183.63005439676388]

______________________________________________________________________________________________________________________________